lib/avo/resources/base.rb

```diff
@@ -0,0 +1,606 @@
+      # We will not overwrite any attributes that come pre-filled in the record.
+      def hydrate_model_with_default_values
+        default_values = get_fields
+          .select do |field|
+            !field.computed && !field.is_a?(Avo::Fields::HeadingField)
+          end
+          .map do |field|
+            value = field.value
+
+            if field.type == "belongs_to"
+
+              reflection = @record._reflections.with_indifferent_access[@params[:via_relation]]
+
+              if field.polymorphic_as.present? && field.types.map(&:to_s).include?(@params[:via_relation_class])
+                # set the value to the actual record
+                via_resource = Avo.resource_manager.get_resource_by_model_class(@params[:via_relation_class])
+                value = via_resource.find_record(@params[:via_record_id])
+              elsif reflection.present? && reflection.foreign_key.present? && field.id.to_s == @params[:via_relation].to_s
+                resource = Avo.resource_manager.get_resource_by_model_class params[:via_relation_class]
+                record = resource.find_record @params[:via_record_id], params: params
+                id_param = reflection.options[:primary_key] || :id
+
+                value = record.send(id_param)
+              end
+            end
+
+            [field, value]
+          end
+          .to_h
+          .select do |_, value|
+            value.present?
+          end
+
+        default_values.each do |field, value|
+          field.assign_value record: @record, value: value
+        end
+      end
+
+      def model_name
+        model_class.model_name
+      end
+
+      def singular_model_key
+        model_class.model_name.singular
+      end
+
+      def record_path
+        resource_path(record: record, resource: self)
+      end
+
+      def records_path
+        resources_path(resource: self)
+      end
+
+      def avatar_field
+        get_field_definitions.find do |field|
+          field.as_avatar.present?
+        end
+      rescue
+        nil
+      end
+
+      def avatar
+        return avatar_field.to_image if avatar_field.respond_to? :to_image
+
+        return avatar_field.value.variant(resize_to_limit: [480, 480]) if avatar_field.type == "file"
+
+        avatar_field.value
+      rescue
+        nil
+      end
+
+      def avatar_type
+        avatar_field.as_avatar
+      rescue
+        nil
+      end
+
+      def form_scope
+        model_class.base_class.to_s.underscore.downcase
+      end
+
+      def has_record_id?
+        record.present? && record_id.present?
+      end
+
+      def id_attribute
+        :id
+      end
+
+      def record_id
+        record.send(id_attribute)
+      end
+
+      def description_attributes
+        {
+          view: view,
+          resource: self,
+          record: record
+        }
+      end
+
+      def entity_loader(entity)
+        instance_variable_get("@#{entity.to_s.pluralize}_loader")
+      end
+    end
+  end
+end
```

Comment:
```suggestion
end

```

Reply:
Changes made added the empty line sorry for that , also @Paul-Bob  would be totally down to write the docs could you just guide me a the repository from what I see is different , so do I raise a seprate pr and link it to this pr ?

But yeah totally down to writing it . 
lmk
Thanks

Reply:
Yes, this is the docs repo https://github.com/avo-hq/docs.avohq.io

You can add a PR there and link it on this PR description, something like:

```
...
Checklist
...
- [x] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
  - Link to docs here <--
...
```

Let me know if you need any help setting up the docs repo

Thank you!